### PR TITLE
Feat/web3 react

### PR DESCRIPTION
**Pull request by mattrussell36**
- Remove custom web3 provider and leverage https://www.npmjs.com/package/@web3-react/core
- Removes web3modal and we can now create our own designed modal.
- Switches components previously using useWeb3 to now use useWeb3React and its properties
- Adds some helper hooks for connecting, disconnecting

New modal looks like this right now (we've lost the logos)
Screen Shot 2022-01-05 at 1 03 48 AM

### CODE REVIEW

File: src/hooks/use-web3.ts [hunk 1]
```diff
@@ -0,0 +1,101 @@
...
+
+  React.useEffect((): any => {
+    const { ethereum } = window as any;
+    if (ethereum && ethereum.on && !active && !error) {
+      const handleConnect = () => {
+        console.log("Handling 'connect' event");
+        activate(injected);
+      };
+      const handleChainChanged = (chainId: string | number) => {
+        console.log("Handling 'chainChanged' event with payload", chainId);
```
Comment (dexturr) [inline src/hooks/use-web3.ts:72]:
```suggestion
```

File: src/hooks/use-web3.ts [hunk 2]
```diff
@@ -0,0 +1,101 @@
...
+}
+
+export function useInactiveListener() {
+  const { active, error, activate } = useWeb3React();
+
+  React.useEffect((): any => {
+    const { ethereum } = window as any;
+    if (ethereum && ethereum.on && !active && !error) {
+      const handleConnect = () => {
+        console.log("Handling 'connect' event");
```
Comment (dexturr) [inline src/hooks/use-web3.ts:68]:
```suggestion
```

File: src/hooks/use-web3.ts [hunk 3]
```diff
@@ -0,0 +1,101 @@
...
+        activate(injected);
+      };
+      const handleChainChanged = (chainId: string | number) => {
+        console.log("Handling 'chainChanged' event with payload", chainId);
+        activate(injected);
+      };
+      const handleAccountsChanged = (accounts: string[]) => {
+        console.log("Handling 'accountsChanged' event with payload", accounts);
```
Comment (dexturr) [inline src/hooks/use-web3.ts:76]:
```suggestion
```

File: src/hooks/use-web3.ts [hunk 4]
```diff
@@ -0,0 +1,101 @@
...
+        if (accounts.length > 0) {
+          activate(injected);
+        }
+      };
+      const handleNetworkChanged = (networkId: string | number) => {
+        console.log("Handling 'networkChanged' event with payload", networkId);
```
Comment (dexturr) [inline src/hooks/use-web3.ts:82]:
```suggestion
```

File: src/components/etherscan-link/etherscan-link.tsx [hunk 5]
```diff
@@ -48,13 +48,13 @@ export const EtherscanLink = ({
...
 }: EtherscanLinkProps) => {
-  const { chainId } = useWeb3();
+  const { chainId } = useWeb3React();
   let hash: string;
   let href: string | null;
   const { t } = useTranslation();
   const { copy, copied } = useCopyToClipboard();
   const linkText = text ? text : t("View on Etherscan (opens in a new tab)");
-  const createLink = etherscanLinkCreator(chainId);
+  const createLink = etherscanLinkCreator(`0x${chainId}` as EthereumChainId);
```
Comment (dexturr) [inline src/components/etherscan-link/etherscan-link.tsx:57]: nitpick (non-blocking): If we are getting the chainId as a number should we change the type to be just a number?

File: src/components/eth-connect-modal/eth-connect-modal.tsx [hunk 6]
```diff
@@ -0,0 +1,56 @@
...
+      isOpen={appState.ethConnectOverlay}
+      onClose={close}
+      transitionDuration={0}
+    >
+      <div className="modal modal--dark">
+        <Modal>
+          <h2>Connect Ethereum wallet</h2>
+          <div>
+            {Object.entries(Connectors).map(([key, connector]) => {
+              if (key === "networkOnly") return null;
```
Comment (dexturr) [inline src/components/eth-connect-modal/eth-connect-modal.tsx:35]: nitpick (non-blocking): magic string

File: src/components/etherscan-link/etherscan-link.tsx (hunk 5 above)
Comment (mattrussell36) [reply src/components/etherscan-link/etherscan-link.tsx:57]: Lets do  this in a follow up pr

**Pull request merged**